### ISSUE TITLE
Making sense of Occlusion output

### ISSUE THREAD

**Issue by maikefer** [pytorch/captum#410]
Hi Captum-Team. I have another understanding-question, I hope I'm not bothering you too much.
Can you specify (or direct me to the explicit lines of code) what exactly is the output of the occlusion attribution? In the tutorials I only find ` "Let us visualize the attribution, focusing on the areas with positive attribution (those that are critical for the classifier's decision):"` and the original paper provides multiple analysis.
Is it plainly computing the difference between whatever the network outputs (for a specified target)? This would mean that determining what exactly negative and positive values mean is dependent on the network output.

**Comment by bilalsal** [comment]
Hi @maikefer 

Yes, the Occlusion algorithm measures the change in the output due to occluding a particular region.
- If the output value drops, this means the region was crucial for that output (e.g. a predicted class), and the attribution assigned to the region will be positive.
- If the output value remains roughly the same, then the output is indifferent to the occluded information.
- If the output value increases, then the occluded area probably contains distracting information, and  the attribution assigned to the region will be negative.

This recipe provides more examples that you might find useful: https://pytorch.org/tutorials/recipes/recipes/Captum_Recipe.html
(here you can see both negative and positive attribution results using Occlusion https://pytorch.org/tutorials/_static/img/thumbnails/cropped/model-interpretability-using-captum.png)

Hope this helps

**Comment by maikefer** [comment]
Hi @bilalsal bilalsal,
thanks for the answer. This helps to some part, but my main question remains unclear. I'm looking for the exact calculation. From your answer I assume it's 

`prediction(no-occlusion) - prediction(occlusion)`

for the target-class specified in the attribution-method. Or is there some more to this calculation? Is the according line of code somewhere here: https://github.com/pytorch/captum/blob/0f971b2c8d7ffc2a12cc5c05387e9c6981b5f9f0/captum/attr/_core/feature_ablation.py#L313? 

So regarding the range of the occlusion-output. If the target-layer would be one right after the softmax, the range of possible occlusion values would be -1 ...1
If it was before a softmax, then you can't hardly say something about the possible range of the values. Are my assumptions here correct?

thanks!

**Comment by bilalsal** [comment]
Hi @maikefer , yes you are right, we take the absolute difference, there is nothing more to this calculation.
The range after softmax is indeed [-1, 1] while before softmax it is unbounded.

Hope this helps

**Comment by maikefer** [comment]
Hi @bilalsal,
if you're taking the **absolute** difference then the range would be [0..1], wouldn't it? Or are you checking for the sign afterwards?
Thanks!

**Comment by bilalsal** [comment]
Apologize, I meant the plain signed difference (without any relative computation).

**Comment by hamsarajan** [comment]
Is it correct to interpret that the difference in output will be the final atribution value? Why is the range of attribution values bounded between [-1, 1]?
